nilearn/glm/_base.py

```diff
@@ -449,8 +476,223 @@ def _generate_filenames_output(
         return self
 
 
+def _generate_statistical_maps(
+    prefix: str,
+    contrasts,
+    contrast_types,
+    generate_bids_name: bool,
+    entities,
+    entities_to_include: list[str],
+):
+    """Return dictionary containing statmap filenames for each contrast.
+
+    statistical_maps[contrast_name][statmap_label] = filename
+    """
+    if not isinstance(contrast_types, dict):
+        contrast_types = {}
+
+    fields = {
+        "prefix": prefix,
+        "suffix": "statmap",
+        "extension": "nii.gz",
```

Comment:
What if we have a SurfaceImage ?

Reply:
yup I noted that in another issue: it's on the todo list

Reply:
but not in this PR that is already too big